ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/EvmToolCommand.java

```diff
@@ -163,23 +186,46 @@ void setBytes(final String optionValue) {
       description = "Number of times to repeat for benchmarking.")
   private final Integer repeat = 0;
 
+  @Option(
+      names = {"-v", "--version"},
+      versionHelp = true,
+      description = "display version info")
+  boolean versionInfoRequested;
+
   static final Joiner STORAGE_JOINER = Joiner.on(",\n");
   private final EvmToolCommandOptionsModule daggerOptions = new EvmToolCommandOptionsModule();
-  private PrintWriter out =
+  PrintWriter out =
       new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out, UTF_8)), true);
+  InputStream in = new ByteArrayInputStream(new byte[0]);
 
-  void parse(final CommandLine.IExecutionStrategy resultHandler, final String[] args) {
+  void execute(final String... args) {
+    execute(System.in, new PrintWriter(System.out, true, UTF_8), args);
+  }
 
-    final CommandLine commandLine = new CommandLine(this);
-    out = commandLine.getOut();
+  void execute(final InputStream input, final PrintWriter output, final String[] args) {
+    final CommandLine commandLine = new CommandLine(this).setOut(output);
+    out = output;
+    in = input;
+
+    // add dagger-injected options
     commandLine.addMixin("Dagger Options", daggerOptions);
 
     // add sub commands here
     commandLine.registerConverter(Address.class, Address::fromHexString);
     commandLine.registerConverter(Bytes.class, Bytes::fromHexString);
     commandLine.registerConverter(Wei.class, arg -> Wei.of(Long.parseUnsignedLong(arg)));
 
-    commandLine.setExecutionStrategy(resultHandler).execute(args);
+    // change negation regexp so --nomemory works
+    commandLine.setNegatableOptionTransformer(
+        new CommandLine.RegexTransformer.Builder()
+            .addPattern("^--no(\\w(-|\\w)*)$", "--$1", "--[no]$1")
+            .addPattern("^--trace.no(\\w(-|\\w)*)$", "--trace.$1", "--trace.[no]$1")
+            .addPattern("^--(\\w(-|\\w)*)$", "--no$1", "--[no]$1")
+            .addPattern("^--trace.(\\w(-|\\w)*)$", "--trace.no$1", "--trace.[no]$1")
```

Comment:
I have no idea what's going on here, maybe a more detailed comment would help

Reply:
It is indeed an incantation, and any deeper explanation than "it makes --nomemory work" involves explaining PicoCLI magic.  I added a link to the PicoCLI docs where they explain that magic.

Reply:
That certainly helps, I didn't clock it was pico magic first time round.